superbench/benchmarks/micro_benchmarks/cuda_decode_performance/AppDecPerf.cpp

```diff
@@ -0,0 +1,454 @@
+            continue;
+        }
+        if (!_stricmp(argv[i], "-total")) {
+            if (++i == argc) {
+                ShowHelpAndExit("-total");
+            }
+            nTotalVideo = atoi(argv[i]);
+            continue;
+        }
+        if (!_stricmp(argv[i], "-multi_input")) {
+            if (++i == argc) {
+                ShowHelpAndExit("-multi_input");
+            }
+            inputFilesListPath = std::string(argv[i]);
+            continue;
+        }
+        if (!_stricmp(argv[i], "-single")) {
+            bSingle = true;
+            continue;
+        }
+        if (!_stricmp(argv[i], "-host")) {
+            bHost = true;
+            continue;
+        }
+        if (!_stricmp(argv[i], "-codec")) {
+            if (++i == argc) {
+                ShowHelpAndExit("-codec");
+            }
+            std::string codecName = std::string(argv[i]);
+            std::transform(codecName.begin(), codecName.end(), codecName.begin(),
+                           [](unsigned char c) { return std::tolower(c); });
+            if (codecMap.find(codecName) != codecMap.end()) {
+                codec = codecMap[codecName];
+            } else {
+                std::cout << "Codec name not found in the map." << std::endl;
+                exit(1);
+            }
+            continue;
+        }
+        ShowHelpAndExit(argv[i]);
+    }
+}
+
+/**
+ *  @brief  Function to create cuda context and initialize decoder
+ */
+OptimizedNvDecoder *InitOptimizedNvDecoder(int i, const CUdevice &cuDevice, CUcontext &cuContext, bool bSingle,
+                                           bool bHost, cudaVideoCodec codec, CUVIDDECODECAPS decodecaps) {
+    if (!bSingle) {
+        ck(cuCtxCreate(&cuContext, 0, cuDevice));
+    }
+    OptimizedNvDecoder *sessionObject = new OptimizedNvDecoder(cuContext, !bHost, codec, decodecaps);
+    sessionObject->setDecoderSessionID(i);
+    return sessionObject;
+}
+
+/**
+ *  @brief  Function to decode a video in a thread and measure the latency
+ */
+double DecodeVideo(size_t i, const std::vector<OptimizedNvDecoder *> &vDec, const char *szInFilePath, int *pnFrame,
+                   std::exception_ptr &ex) {
+    try {
+        OptimizedNvDecoder *pDec = vDec[i];
+        auto start = std::chrono::high_resolution_clock::now();
+        DecProc(pDec, szInFilePath, pnFrame, ex);
+        auto end = std::chrono::high_resolution_clock::now();
+        auto elapsedTime = std::chrono::duration_cast<std::chrono::milliseconds>(end - start).count();
+        std::cout << "Decode finished --"
+                  << " duration:" << elapsedTime << " frames:" << *pnFrame << std::endl;
+        return elapsedTime / 1000.0f;
+    } catch (const std::exception &e) {
+        std::cerr << "Exception in deocding: " << e.what() << std::endl;
+        return 0;
+    }
+}
+
+/**
+ *  @brief  Function to read the video paths from a file
+ */
+std::vector<std::string> ReadMultipleVideoFiles(std::string filepath) {
```

Comment:
>std::string [](http://example.com/codeflow?start=48&length=11)

const std::string&

Reply:
updated in ReadMultipleVideoFiles(const std::string &filepath)  and GenerateTotalFileList(const std::string &inputFilesListPath, int nTotalVideo,
                                               const char *szInFilePath)